src/components/PerPartyMapContainer.js

```diff
@@ -55,12 +44,40 @@ function getMapData(summaryState, partyId) {
     }
     return [
       ...zones.map((zone, i) => {
-        const { candidate, zoneStats } = getSeatDisplayModel(summary, zone)
-        const onMap = candidate && candidate.partyId === partyId
+        const { candidate: winningCandidate, zoneStats } = getSeatDisplayModel(
+          summary,
+          zone
+        )
+        const win = winningCandidate && +winningCandidate.partyId === +partyId
+        const sentCandidate = perPartyModel.getCandidate(
+          zone.provinceId,
+          zone.no
+        )
+        let opacity = 1
+        let winningPartyId = "nope"
+        let complete = false
+        const interpolate = (value, min = 0, max = 1) =>
```

Comment:
use `d3.scaleLinear` or more specifically `import { scaleLinear } from 'd3-scale'` will save you many lines. 

Also worth considering `scaleQuantize` instead to provide steps of colors instead of continuous range.

Reply:
Thanks for your suggestion — will do that.